CONTRIBUTING_WITH_GIT2R.Rmd

```diff
@@ -0,0 +1,551 @@
+
+```{r add_stuff_shadow, echo = FALSE}
+x <- file.copy(paste0(fork_pgi, "/TEMPLATE.Rmd"), 
+               paste0(fork_pgi, "/use/2015-12-16-spatial-stats.Rmd"))
+x <- file.create(paste0(fork_pgi, "/data/spatial_data.csv"))
+set.seed(999)
+m <- matrix(rnorm(20), nrow = 5, ncol = 2, dimnames = list(NULL, c("x", "y")))
+write.table(m, file = paste0(fork_pgi, "/data/spatial_data.csv"), sep = ",", 
+            row.names = FALSE, col.names = TRUE)
+```
+
+```{r status}
+status(repo)
+```
+
+Since we've just added the files, we can use `add()` to tell git we want to
+stage them for committing.
+
+```{r add}
+add(repo, "*") # adding all new or changed files
+status(repo)
+```
+
+When we have done this, we can commit to our changes by using `commit()`. A
+commit is basically placing a record of what you did and, importantly, why you
+did it. Your commit message should record this. Commit messages should be able
+to fit on a single line. Optionally, if you want to be able to add more detail,
+you can enter more lines below the message:
+
+```{r commit}
+msg <- " started spatial stats vignette and added data
+
+I copied the template to the use folder and placed
+spatial data in the data folder.
+"
+commit(repo, msg)
+status(repo) # should return nothing
+repo
+```
+
+You can also see a summary of what you just did by looking at the summary of
+your commit:
+
+```{r summarycommit}
+summary(commits(repo)[[1]])
+```
+
+
+Now you can see that your commit message was recorded in the repository. Since
+we see that there is a **Remote** associated with this fork, we can simply use
+`push()` to push our changes up to the fork.
+
+```{r pushit, eval = FALSE}
+push(repo)
+```
+
+# Conclusions
+
+From this tutorial, you should have learned how to:
+
+ - fork a repository on GitHub
+ - clone that fork
+ - set up credentials for *git2r*
+ - update your master branch from NESCent
+ - create a new branch
+ - add and commit changes
+ - push changes
+
+The skills demonstrated in this tutorial are not exclusive for **popgenInfo**,
+but they can be used when writing up your own reports or analyses. Using git may
+not feel comfortable at this moment, but just like it takes practice to learn
+how to keep a lab notebook, with practice, you will become comfortable using git
+to keep track of your workflows, making your science more reproducible.
+
+# What's Next?
+
+Now that you've successfully set up git, created a new branch, and created a new
+vignette, it's time to fill it with content. You can work in your vignette like
+you would write in any other Rmarkdown document and keep track of your changes
+like we showed above by **add**ing and **commit**ting your changes followed by
+pushing these up to your fork. Once you are finished with all of your changes,
+you can create a pull request from your GitHub fork to NESCent. After it's
+reviewed by at least two of our maintainers, it gets merged into NESCent. From
```

Comment:
It _might_ get merged. But since the review isn't just rubber-stamping, it may also result in feedback and requests for fixes or improvements. In fact, I suppose that would be the "normal" case, and thus it's perhaps better to include that in the flow being explained.